apps/src/music/views/MusicView.jsx

```diff
@@ -197,26 +197,28 @@ class UnconnectedMusicView extends React.Component {
       return;
     }
 
-    // Stop all when_run sounds that are still to play, because if they
-    // are still valid after the when_run code is re-executed, they
-    // will be scheduled again.
-    this.stopAllSoundsStillToPlay();
-
-    // Also clear all when_run sounds from the events list, because it
-    // will be recreated in its entirely when the when_run code is
-    // re-executed.
-    this.player.clearWhenRunEvents();
-
-    this.executeSong();
-
-    this.analyticsReporter.onBlocksUpdated(
-      this.musicBlocklyWorkspace.getAllBlocks()
-    );
-
-    // This is a way to tell React to re-render the scene, notably
-    // the timeline.
-    this.setState({updateNumber: this.state.updateNumber + 1});
+    const codeChanged = this.compileSong();
+    if (codeChanged) {
```

Comment:
nit: invert logic and early return?

Reply:
Could go either way on this, but a motivator for the current approach is that there remains one last call to do whether we execute again or not, and it's the workspace save.  I liked the idea of not doing that until after we'd done everything else, whether we were going to re-execute or not.